garage/logger/snapshotter.py

```diff
@@ -78,3 +82,43 @@ def save_itr_params(self, itr, params):
             if file_name:
                 with open(file_name, 'wb') as file:
                     pickle.dump(params, file)
+
+    def load(self, itr='last'):
+        r"""Load one snapshot of parameters from disk.
+
+        Args:
+            itr(int or string): Iteration to load.
+                Can be an integer, 'last' or 'first'.
+
+        Returns:
+            dict: Loaded snapshot
+
+        """
+        if itr is None:
+            itr = 'last'
+
+        if isinstance(itr, int) or itr.isdigit():
+            snapshot_file = osp.join(self._snapshot_dir,
+                                     'itr_{}.pkl'.format(itr))
+        else:
+            if not (itr in ('last', 'first')):
+                raise ValueError(
+                    "itr should be an integer or 'last' or 'first'")
+
+            snapshot_file = osp.join(self._snapshot_dir, 'params.pkl')
+            if not osp.isfile(snapshot_file):
+                files = [
+                    f for f in listdir(self._snapshot_dir)
+                    if f.endswith('.pkl')
+                ]
+                if not files:
+                    raise FileNotFoundError(errno.ENOENT,
+                                            os.strerror(errno.ENOENT),
+                                            '*.pkl file')
+                files.sort(key=lambda f: int(re.findall(r'\d+', f)[0]))
+                snapshot_file = files[0] if itr == 'first' else files[-1]
+                snapshot_file = osp.join(self._snapshot_dir, snapshot_file)
+
+        assert osp.isfile(snapshot_file)
+
+        return pickle.load(open(snapshot_file, 'rb'))
```

Comment:
if you are going to always load with pickle, you need to make sure you're not saving with joblib

Reply:
This is to keep consistent with `Snapshotter.save_itr_params()` that uses pickle. I found that we have mixing uses of pickle and joblib to do dump and load in our codebase. This is not good. Since joblib is faster dealing with ndarray, should we just get rid of pickle?

Reply:
the main problem is that something saved with `joblib.dump()` cannot be loaded with `pickle.load()`.

I think we should stick with vanilla pickle until we have evidence that joblib is necessary. This doesn't mean we can't use joblib (for instance) in the sampler, because those bytes are never committed to disk.

Reply:
Alternatively, as long as joblib is used for both save and load, it's fine.

Reply:
joblib is used in unpickling now.